Jenkinsfile

```diff
@@ -0,0 +1,74 @@
+#!groovy
+
+pipeline {
+  agent {
+    docker {
+        image 'dealii/dealii:v8.5.1-gcc-mpi-fulldepscandi-debugrelease'
+	label 'has-docker'
+    }
+  }
+
+  parameters {
+    booleanParam(defaultValue: false, description: 'Is the pull request approved for testing?', name: 'TRUST_BUILD')
+  }
```

Comment:
I haven't looked at the Jenkins settings for who can trigger tests, but I would hope that it would be entirely on the Jenkins side. Since the Jenkinsfile is in the repository, this is just as susceptible to malicious pull requests as everything else.

Reply:
That's interesting. I wonder why Jenkins decided to do it that way. I'm not sure how much security you can get from using a trusted Jenkinsfile to build untrusted code.

It looks like you try to avoid that problem by erroring with this `TRUST_BUILD` whitelist, but you're still running the indentation checker on untrusted code.

Also, instead of killing the build with an error and telling the submitter to contact an admin, would it be better to use the Jenkinsfile [input directive](https://jenkins.io/doc/book/pipeline/syntax/#input)? ([Interface example](https://jenkins.io/doc/tutorials/create-a-pipeline-in-blue-ocean/#add-a-final-deliver-stage-to-your-pipeline), scroll down a bit)

Reply:
> but you're still running the indentation checker on untrusted code.

That is fine and I have been doing that since forever. Except that using ``doc/indent`` from untrusted code is not smart. Maybe reimplement this script? ;-)

> to use the Jenkinsfile input directive?

Yes, that is also an option except that it was somewhat finicky when I tried it. I also haven't figured out how to do this input check before a node was allocated. This way pull requests will block the CI servers.